packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -677,6 +678,22 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
     } else {
       throwBadResponse(stateP, res);
     }
+  } else if (stateP.controlState === 'CHECK_CLUSTER_ROUTING_ALLOCATION') {
+    const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
+    if (Either.isRight(res)) {
+      return {
+        ...stateP,
+        controlState: 'CHECK_UNKNOWN_DOCUMENTS',
+      };
+    } else {
+      const left = res.left;
+      if (isTypeof(left, 'incompatible_cluster_routing_allocation')) {
+        const retryErrorMessage = `[${left.type}] Incompatible Elasticsearch cluster settings detected. Remove the persistent and transient Elasticsearch cluster setting 'cluster.routing.allocation.enable' or set it to a value of 'all' to allow migrations to proceed. Refer to ${stateP.migrationDocLinks.routingAllocationDisabled} for more information on how to resolve the issue.`;
+        return delayRetryState(stateP, retryErrorMessage, stateP.retryAttempts);
```

Comment:
tbh I'm surprised we were retrying in case of `incompatible_cluster_routing_allocation`, given it's not really something that could be transient, but the behavior was already there, so I get we'd rather keep it.

Reply:
++ we can open a separate issue to investigate this, and/or ask Rudolf to chime in when he's back.